libraries/platforms/dromajo-vcs/bsg_manycore_simulator.cpp

```diff
@@ -0,0 +1,624 @@
+
+    // Poll for packets to be received
+    bsg_pr_dbg("Checking for packets to receive\n");
+    err = dromajo_receive_packet();
+    if (err != HB_MC_SUCCESS)
+      return err;
+  }
+
+  // Advance time 1 unit
+  advance_time();
+  return HB_MC_SUCCESS;
+}
+
+////////////////////////////// Interacting with VCS //////////////////////////////
+
+/*
+ * get_argc
+ * Given a string, determine the number of space-separated arguments
+ * @params[in] args - Pointer to a character array that holds the arguments
+ * @returns the number of arguments
+ */
+static int get_argc(char * args){
+        char *cur = args, prev=' ';
+        int count = 1;
+        while(*cur != '\0'){
+                if((prev == ' ') && (prev != *cur)){
+                        count ++;
+                }
+                prev = *cur;
+                ++cur;
+        }
+        return count;
+}
+
+/*
+ * get_argv
+ * Given a string, retrieves the space-separated arguments
+ * @params[in] args - Pointer to a character array that holds the arguments
+ * @params[in] argc - Number of arguments
+ * @params[in] argv - Pointer to an array of strings that will hold the different
+ * arguments
+ */
+static void get_argv(char * args, int argc, char **argv){
+        int count = 0;
+        char *cur = args, prev=' ';
+
+        // First parse the path name. This is not in the argument string because
+        // VCS doesn't provide it to us. Instead, we "hack" around it by reading
+        // the path from 'proc/self/exe'. The maximum path-name length is 1024,
+        // with an extra null character for safety
+        static char path[1025] = {'\0'};
+
+        readlink("/proc/self/exe", path, sizeof(path) - 1);
+        argv[0] = path;
+        count ++;
+
+        // Then we parse the remaining arguments. Arguments are separated by N
+        // >= 1 spaces. We only register an argument when the previous character
+        // was a space, and the current character is not (so that multiple
+        // spaces don't count as multiple arguments). We replace spaces with
+        // null characters (\0) so that each argument appears to be an
+        // individual string and can be used later, by argparse (and other
+        // libraries)
+        while(*cur != '\0'){
+                if((prev == ' ') && (prev != *cur)){
+                        argv[count] = cur;
+                        count++;
+                }
+                prev = *cur;
+                if(*cur == ' ')
+                        *cur = '\0';
+                cur++;
+        }
+}
+
+/*
+ * vcs_main
+ * The main function for the x86 host. It initializes all devices and controls
+ * the time in simulation. It handles all DPI calls and passes packets to and from
+ * Dromajo during execution
+ * @params[in] argc - Number of arguments
+ * @params[in] argv - Pointer to an array of string arguments
+ * @returns
+ */
+int vcs_main(int argc, char **argv) {
+  // Push command-line arguments into global variables
```

Comment:
Is there no other way to pass them?

Reply:
I don't think so or rather I don't know if it is with the current state of the system. Once, BlackParrot argument passing is fixed, we can deprecate the start code in this repository (as previously discussed in another review comment) and just push the contents to the stack. Until then, I thought this was reasonable.

Reply:
I recommend passing them as a field in SimulationWrapper since I think that is where they are accessed, and assignment happens immediately before construction.

Reply:
What do you mean when you say "before construction"? I have it assigned in the initialization process for the simulator. Right after the DPI and Dromajo pointers are initialized before the simulation begins.